src/app/energy-summary-list/calculate-energy-data-with-battery.ts

```
@@ -18,75 +19,62 @@ import {
-    totalAmount: 0,
-  };
+  let fedIntoGrid = 0; // wyslane do sieci
 
-  let calculatedG11ProductionAmount: TariffEnergyAmountData = {
-    amount: { [TariffZone.Day]: 0 },
-    totalAmount: 0,
-  };
-  let calculatedG12ProductionAmount: TariffEnergyAmountData = {
-    amount: { [TariffZone.Day]: 0, [TariffZone.Night]: 0 },
+  const initialTariffEnergyAmountData = (zones: TariffZone[]): TariffEnergyAmountData => ({
+    amount: zones.reduce((acc, zone) => ({ ...acc, [zone]: 0 }), { [TariffZone.Day]: 0 }),
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)